libmodernize/main.py

```diff
@@ -17,6 +18,35 @@
 from libmodernize import __version__
 from libmodernize.fixes import lib2to3_fix_names, six_fix_names, opt_in_fix_names
 
+
+class LFPreservingRefactoringTool(StdoutRefactoringTool):
+    """ https://github.com/python-modernize/python-modernize/issues/121 """
+    def write_file(self, new_text, filename, old_text, encoding):
+        # detect linefeeds
+        lineends = {'\n':0, '\r\n':0, '\r':0}
+        lines = []
+        for line in open(filename, 'rb'):
```

Comment:
Fair point. There are ways around that, like using `errors='ignore'` to skip characters that can't be decoded, but it may be easier to use binary mode here. In that case, though, you'll need to update the checks below to use bytes, e.g. `if line.endswith(b'\r\n')`.